--- app_meetme_orig.c	2007-03-09 16:19:50.000000000 -0500
+++ app_meetme.c	2007-03-09 16:01:54.000000000 -0500
@@ -122,6 +122,16 @@
 "      'M' -- Mute conference\n"
 "      'n' -- Unmute entire conference (except admin)\n"
 "      'N' -- Mute entire conference (except admin)\n"
+"      'S' -- Raise entire conference speaking volume\n"
+"      's' -- Lower entire conference speaking volume\n"
+"      'V' -- Raise entire conference listening volume\n"
+"      'v' -- Lower entire conference listening volume\n"
+"      'T' -- Raise one user's talk volume\n"
+"      't' -- Lower one user's talk volume\n"
+"      'U' -- Raise one user's listen volume\n"
+"      'u' -- Lower one user's listen volume\n"
+"      'R' -- Reset all user's volume settings\n"
+"      'r' -- Reset one user's volume settings\n"
 "";
 
 #define CONFIG_FILE_NAME "meetme.conf"
@@ -2114,6 +2124,61 @@
 					ast_log(LOG_NOTICE, "Specified User not found!\n");
 				}
 				break;
+			case 118: /* v: Lower all users listen volume */
+AST_LIST_TRAVERSE(&cnf->userlist, user, list) {
+					tweak_listen_volume(user, VOL_DOWN);
+				}
+				break;
+			case 86: /* V: Raise all users listen volume */
+AST_LIST_TRAVERSE(&cnf->userlist, user, list) {
+					tweak_listen_volume(user, VOL_UP);
+				}
+				break;
+			case 115: /* s: Lower all users speaking volume */
+AST_LIST_TRAVERSE(&cnf->userlist, user, list) {
+					tweak_talk_volume(user, VOL_DOWN);
+				}
+				break;
+			case 83: /* S: Raise all users speaking volume */
+AST_LIST_TRAVERSE(&cnf->userlist, user, list) {
+					tweak_talk_volume(user, VOL_UP);
+				}
+				break;
+			case 82: /* R: Reset all volume levels */
+AST_LIST_TRAVERSE(&cnf->userlist, user, list) {
+					reset_volumes(user);
+				}
+				break;
+			case 114: /* r: Reset user's volume level */
+				if (user)
+					reset_volumes(user);
+				else
+					ast_log(LOG_NOTICE, "Specified User not found!");
+				break;
+			case 85: /* U: Raise user's listen volume */
+				if (user)
+					tweak_listen_volume(user, VOL_UP);
+				else
+					ast_log(LOG_NOTICE, "Specified User not found!");
+				break;
+			case 117: /* u: Lower user's listen volume */
+				if (user)
+					tweak_listen_volume(user, VOL_DOWN);
+				else
+					ast_log(LOG_NOTICE, "Specified User not found!");
+				break;
+			case 84: /* T: Raise user's talk volume */
+				if (user)
+					tweak_talk_volume(user, VOL_UP);
+				else
+					ast_log(LOG_NOTICE, "Specified User not found!");
+				break;
+			case 116: /* t: Lower user's talk volume */
+				if (user) 
+					tweak_talk_volume(user, VOL_DOWN);
+				else 
+					ast_log(LOG_NOTICE, "Specified User not found!");
+				break;
 			}
 
 			dispose_conf(cnf);
